python/cudf/cudf/core/column/column.py

```diff
@@ -2719,8 +2719,16 @@ def as_column(
             return as_column(arbitrary, dtype=dtype, nan_as_null=nan_as_null)
         elif arbitrary.dtype.kind in "biuf":
             from_pandas = nan_as_null is None or nan_as_null
+            try:
+                pa_array = pa.array(arbitrary, from_pandas=from_pandas)
+            except pa.ArrowNotImplementedError:
```

Comment:
Can we check if an array is byte swapped? Instead of relying on generic not implemented error?

Reply:
Sure, switch to check if the byte order is non native.